internal/literals/literals.go

```diff
@@ -97,74 +93,48 @@ func Obfuscate(file *ast.File, info *types.Info, fset *token.FileSet, ignoreObj
 				return true
 			}
 
-			data := make([]byte, 0, len(x.Elts))
+			data := make([]byte, 0, len(node.Elts))
 
-			for _, el := range x.Elts {
-				lit, ok := el.(*ast.BasicLit)
-				if !ok {
+			for _, el := range node.Elts {
+				elType := info.Types[el]
+
+				if elType.Value == nil || elType.Value.Kind() != constant.Int {
 					return true
 				}
-				var value byte
-				if lit.Kind == token.CHAR {
-					val, err := strconv.Unquote(lit.Value)
-					if err != nil {
-						panic(fmt.Sprintf("cannot unquote character: %v", err))
-					}
-
-					value = byte(val[0])
-				} else {
-					val, err := strconv.ParseUint(lit.Value, 0, 8)
-					if err != nil {
-						panic(fmt.Sprintf("cannot parse integer: %v", err))
-					}
-
-					value = byte(val)
+
+				value, ok := constant.Uint64Val(elType.Value)
+				if !ok {
+					return true
```

Comment:
under what circumstances could this happen? I see:

> A special Unknown value may be used when a value is unknown due to an error. Operations on unknown values produce unknown values unless specified otherwise. 

> Uint64Val returns the Go uint64 value of x and whether the result is exact; x must be an Int or an Unknown. If the result is not exact, its value is undefined. If x is Unknown, the result is (0, false). 

So presumably this should never happen for us, as we require the code to typecheck correctly. So I would probably panic here just like any other unexpected error, as opposed to silently disabling the obfuscation.